src/substrate/useSubstrate.js

```diff
@@ -0,0 +1,64 @@
+import { useContext, useEffect, useCallback } from 'react';
+import { ApiPromise, WsProvider } from "@polkadot/api";
+import { web3Accounts, web3Enable } from "@polkadot/extension-dapp";
+import keyring from "@polkadot/ui-keyring";
+
+import config from "../config";
+import { SubstrateContext } from './SubstrateContext';
+
+const useSubstrate = () => {
+  let [state, dispatch] = useContext(SubstrateContext);
+
+  // `useCallback` so that returning memoized function and not created
+  //   everytime, and thus re-render.
+  const connect = useCallback(async() => {
+    let { api, socket, types } = state;
+    if (api) return;
+
+    const provider = new WsProvider(socket);
+
+    try {
+      api = await ApiPromise.create({ provider, types });
+      dispatch({ type: 'CONNECT', payload: api });
+      api.isReady.then(() => dispatch({type: 'CONNECT_SUCCESS'}));
+
+    } catch(e) {
+      console.log(e);
+      dispatch({ type: 'CONNECT_ERROR' });
+    }
+    // eslint-disable-next-line
+  }, [state.api, state.socket]);
+
+  // hook to get injected accounts
+  const loadAccounts = useCallback(async() => {
+
+    // Ensure the method only run once.
+    if (state.keyringState) return;
+
+    try {
+      await web3Enable(config.APP_NAME);
+      let allAccounts = await web3Accounts();
+      allAccounts = allAccounts.map(({ address, meta }) =>
+        ({ address, meta: { ...meta, name: `${meta.name} (${meta.source})` }}));
+
+      keyring.loadAll({ isDevelopment: config.DEVELOPMENT_KEYRING }, allAccounts);
+      dispatch({ type: 'SET_KEYRING', payload: keyring });
+    } catch (e) {
+      console.log(e);
+      dispatch({ type: 'KEYRING_ERROR' });
+    }
+    // eslint-disable-next-line
```

Comment:
```
Line 49:  React Hook useCallback has missing dependencies: 'dispatch' and 'state.keyringState'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
```

Reply:
same logic as above. All the more, reading the doc `web3Enable()` is a function that must only be called for one time. It returns error when called the 2nd time. So it is something to be done once. 

But maybe there is a better way to handle it....

Reply:
fixed :)